### DISCUSSION TITLE
documentation for svelte / sapper ssr?

### DISCUSSION THREAD

**Discussion by jdgamble555** [urql-graphql/urql#1405] [Q&A] [answered]
Does **urql** support svelte (with sapper) **ssr**?

I am not sure if the documentation was left out, or if it is unsupported.  I believe a lot of people think they are getting ssr support, when really it is just running on the front end with sapper.  

Thanks,
J

**Comment by jdgamble555** [comment, marked as answer]
@kitten Is **urql** supported in Sapper?

**Reply by JoviDeCroock** [reply]
Hey, why wouldn't it be supported? Sapper afaik is still Svelte but wraps it so it can run on the server. Personally haven't done the whole ssr story on Svelte but judging from https://github.com/timhall/svelte-apollo/issues/9 it's more of a manual user-facing concept to make the server-client data-transfer.

This would be similar to how we use the ssrExchange in React/Vue.

Not exactly used to Svelte but can look into making an example

**Reply by JoviDeCroock** [reply]
Started a small sandbox but getting so many issues with running it into CSB but the general idea is

- Preload in ssr
- Restore with the `ssrExchange` (note that the initialState location is probably wrong currently)
- Query on the client again and it should be synchronous

https://codesandbox.io/s/nifty-sound-rjrj0?file=/src/urqlClient.js

**Reply by jdgamble555** [reply]
Nice! Been trying to figure this out for a while now, as not many other people got this right.  The key for me here was the `preload` and running the `@urql/core` instead of `@urql/svelte` when doing mutations.  Also had to install `isomorphic-unfetch` as said randomly in these forms...

Here is my working repository: https://github.com/jdgamble555/sapper-urql-ssr

**Reply by JoviDeCroock** [reply]
Currently you're not using the ssrExchange though, this would help you embed said entity inside of the cache between server-delivery and client hydration.

**Reply by jdgamble555** [reply]
Do I need it if I am using subscriptions on the front end? It is not cached from my understanding, hence why you can’t use optimistic updates etc?
